include/functions/math.hpp

```diff
@@ -125,6 +124,60 @@ BigInt sqrt(const BigInt& num) {
     return sqrt_current;
 }
 
+/*
+   is_probable_prime
+   -----------------
+   Returns bool that indicates the primality of a BigInt based 
+   on primality testing using Rabin-Miller testing
+   NOTE: certainty determines how accurate test is and must be greater than 0
+   			
+ */
+
+bool BigInt::is_probable_prime(size_t certainty) const {
+    BigInt num = this->value;
```

Comment:
This will set `num` to the absolute value of this object (same as `num = abs(*this)`). Is that desired behaviour when a number is negative? (I'm just confirming since I'm not sure myself.)

Reply:
According to WolframAlpha, negative of a prime is NOT a prime.

Reply:
Alright, thanks I've fixed that issue